tests/globus_test.py

```diff
@@ -0,0 +1,394 @@
+from unittest import *
+import inspect
+import os
+import telnetlib
+import hashlib
+import zlib
+from libs.execute import *
+from libs.command import *
+from libs.utility import *
+from datetime import datetime
+from ftplib import FTP, error_perm
+
+def get_telnet_checksum_command(checksum_type, filename):
+    cksm_command = bytearray(b'cksm ')
+    cksm_command.extend(checksum_type.encode())
+    cksm_command.extend(b' 0 1 ')
+    cksm_command.extend(filename.encode())
+    cksm_command.extend(b'\r\n')
+    return cksm_command
+
+# convert the response (example b'213 77de104b6b11b23d4a6f22adb044884c\r\n') to
+# a string and grab only the checksum
+def parse_telnet_checksum_response_as_string(response):
+    return response.decode('utf-8').split(' ')[1].splitlines()[0]
+
+class Globus_Test(TestCase):
+
+    hostname = os.environ["HOSTNAME"] 
+
+    def __init__(self, *args, **kwargs):
+        super(Globus_Test, self).__init__(*args, **kwargs)
+
+    def setUp(self):
+        pass
+
+    def tearDown(self):
+        pass
+
+    def test_resource_mapfile(self):
+        # the resource mapfile was already set up by the docker scripts
```

Comment:
Is there a way to check this in the test (e.g. stat some file, ask Globus, etc.)? Could be useful for understanding that particular failure should it arise.

Reply:
I can add that test.

Reply:
I added a test to check for error with that mapfile is not there.  I think this is more of a pure Globus test but it won't hurt.

Reply:
Ah, I should have been more clear about my intention. The new test is fine, I guess, but I was thinking that such an assertion could be placed at the top of this test to ensure that the pre-requisite indicated by this comment is being met before attempting to run the test. That way, it is clear *why* the test failed (or could not be run, or whatever we decide to do with the fact that the resource mapfile wasn't set up properly) rather than running ahead in the test and failing with some possibly cryptic error.

I had something like this in mind, from the authentication tests which assume certain users with certain passwords exist before the test runs: https://github.com/irods/irods/blob/5e5057bdfff711163dcc6d5a26315997da1058a9/scripts/irods/test/test_auth.py#L33-L42

This might be overkill as a start, but just thought I would suggest it